src/core/internal/fetch/fetchClient.ts

```diff
@@ -27,24 +32,19 @@ export class FetchClient implements HttpClient {
         return this.request(url, 'PATCH', config, data);
     }
 
-    private async request(url: string, method: string, config?: { headers: any }, data?: any) {
-        let body = undefined;
-        if (!!data) {
-            if (typeof data === 'object') {
-                body = JSON.stringify(data);
-                config = config || { headers: {} };
-                config.headers = config.headers || {};
-                config.headers['Content-Type'] = 'application/json';
-            } else {
-                body = data;
-            }
-        }
-        await this.waitInQueue();
+    private async request(url: string, method: string, config?: IRequestConfig, data?: any) {
+        const reqConfig = config || {};
+        const reqData = data || {};
 
+        await this.waitInQueue();
         return fetch(url, {
-            method: method,
-            headers: !!config ? config.headers : {},
-            body: body
+            headers: {
+                ...reqConfig.headers,
+                'Content-Type': 'application/json'
```

Comment:
Will not work properly. 'Content-Type' header can be overridden.

Reply:
I thought that we use just json type, but okay will fix it